### PR TITLE
HAC-1618: Add create workspace modal

### PR DESCRIPTION
**Pull request by kdoberst**
This is the initial PR to add a modal to create a new workspace. 

Adds a "Create workspace" button on the list of workspaces page. 

**Screen video**

https://user-images.githubusercontent.com/82059948/187110966-7ffe795d-fbf6-4f50-a859-007b246af652.mov

Directions on how to setup the local environment - https://docs.google.com/document/d/1hzoVBBi_694WfmoudtdS9mg1ZWvtox-S7fPYkDnBn5Q

NOTE: There are numerous changes that will be needed to make this fully functional:
 - Addition of the ability to add action buttons (like "Create workspace") to the ListView of the dynamic-plugin-sdk-utils
 - A details page where the user will be directed after a workspace has been successfully created.
 - Check to see if a user has the ability to create a workspace before displaying the create button.
 
 NOTE:  There appears to be an issue with deleting workspaces.  If you create a workspace, deleted it, and try to create another workspace with the same name shortly after the delete, the final creation will fail.  This is reflected in the video above.

Addresses: [HAC-1618](https://issues.redhat.com//browse/HAC-1618)

### CODE REVIEW

File: src/components/WorkspaceAdd/WorkspaceAddButton.tsx [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        ]}
+      >
+        {error ? <Alert variant="danger" isInline title={error} role="alert" /> : null}
+        {loading ? 'Loading ....' : null}
+        {!error && !loading ? (
+          <Form>
+            <FormGroup
+              label="Name"
+              isRequired
+              helperTextInvalid={nameExists ? 'This name already exists' : 'Must only contains letters, numbers, and dashes'}
```
[the remark this answers is not in the document]
Comment (kdoberst) [reply src/components/WorkspaceAdd/WorkspaceAddButton.tsx]: Yep - this isn't right.  Changed

File: src/components/WorkspaceAdd/WorkspaceAddButton.tsx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+  const [workspaceName, setWorkspaceName] = React.useState('');
+  const [workspaceType, setWorkspaceType] = React.useState(workspaceTypeDefault);
+  const [isValidName, setIsValidName] = React.useState(ValidatedOptions.default);
+  const [nameExists, setNameExists] = React.useState(false);
+  const [error, setError] = React.useState('');
+  const [loading, setLoading] = React.useState(false);
+
+  const createWorkspace = () => {
+    setLoading(true);
+    // TODO: We need to extend K8sResourceCommon to include the params below
```
[the remark this answers is not in the document]
Comment (kdoberst) [reply src/components/WorkspaceAdd/WorkspaceAddButton.tsx]: OK - think I figured out what was going on.  Added what was needed to make typescript happy.

**Review by vidyanambiar — APPROVED** [review]
/lgtm

**Pull request merged**